lib/vanagon/project.rb

```diff
@@ -82,6 +82,32 @@ def get_files
       files.flatten.uniq
     end
 
+    # Returns a filtered out set of components only including those
+    # components necessary to build a specific component. This is a
+    # recursive function that will call itself until it gets to a
+    # component with no build requirements
+    #
+    # @param name [String] name of component to add. must be present in configdir/components and named $name.rb currently
+    def filter_component(name)
+      components_to_include = []
+      filter_component = get_component(name)
+      return nil if filter_component.nil?
+      components_to_include << filter_component
+      filter_component.build_requires.each do |build_requirement|
+        unless get_component(build_requirement).nil?
+          components_to_include += filter_component(build_requirement)
+        end
+      end unless filter_component.build_requires.nil?
+      components_to_include
+    end
+
+    def get_component(name)
+      unless @components.select { |comp| comp.name.to_s == name.to_s }.nil?
+        return @components.select { |comp| comp.name.to_s == name.to_s }.pop
+      end
+      nil
+    end
```

Comment:
`#select` returns a new array with the value of whatever matched in the block, or empty otherwise. Checking for `empty?` would be more appropriate than `nil?` since an empty array is still truthy.

```ruby
>> components = [ { name: 'shazam' }, { name: 'mary marvel' }, { name: 'black adam' } ]
=> [
    [0] {
        :name => "shazam"
    },
    [1] {
        :name => "mary marvel"
    },
    [2] {
        :name => "black adam"
    }
]
>> components.select { |comp| comp[:name].to_s == 'shazam' }
=> [
    [0] {
        :name => "shazam"
    }
]
>> components.select { |comp| comp[:name].to_s == 'spider-man' }
=> []
>> components.select { |comp| comp[:name].to_s == 'spider-man' }.empty?
=> true
>> components.select { |comp| comp[:name].to_s == 'spider-man' }.nil?
=> false
```